runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaPublishFactory.java

```diff
@@ -1329,10 +1312,24 @@ private void onKafkaWindow(
 
             assert initialAck <= initialSeq;
 
-            if (delegate != null)
+            if (wasOpen)
+            {
+                delegate.doMqttWindow(traceId, authorization, budgetId, padding, capabilities);
+            }
+            else if (delegate.qos < MqttQoS.EXACTLY_ONCE.value())
```

Comment:
These `if` conditions can be combined via `||` since the behavior for each is identical, agree?